spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/web/WebMvcAutoConfiguration.java

```diff
@@ -227,6 +228,15 @@ public LocaleResolver localeResolver() {
 		}
 
 		@Bean
+		@ConditionalOnMissingBean
+		@ConditionalOnProperty(prefix = "spring.mvc", name = "default-locale")
+		public LocaleResolver defaultLocaleResolver() {
```

Comment:
There is no exclusion. if you have both `spring.mvc.locale` and `default-locale` on the classpath you end up with two `LocalResolver` beans which is probably not what you want. I guess you need to move those to two inner class that you import explicitly. Look at `DataSourceAutoConfiguration` that has a package protected class for the connection pools we support. 

We should have a test that if both these properties are set, `spring.mvc.locale` is used. Also I am wondering if we shouldn't have a boolean flag instead. Let me raise that on the issue.